document/src/main/java/com/yahoo/document/annotation/SimpleIndexingAnnotations.java

```diff
@@ -0,0 +1,149 @@
+// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.document.annotation;
+
+import com.yahoo.document.datatypes.StringFieldValue;
+
+/**
+ * Lightweight representation of TERM annotations for indexing.
+ * Uses flat arrays instead of object graphs for 80-90% memory reduction.
+ * NOT part of public API - internal optimization for indexing performance.
+ *
+ * This class can only represent simple TERM annotations with:
+ * - Position (from, length)
+ * - Optional term override (when term differs from substring)
+ *
+ * Memory footprint: ~8 bytes + arrays vs ~300 bytes per annotation with full SpanTree.
+ *
+ * @author havardpe
+ */
+public final class SimpleIndexingAnnotations {
```

Comment:
Consider putting this in a sub-package ".internal" that has @ExportPackages but not @PublicApi